web/src/actions/editApd/overview.js

```diff
@@ -1,6 +1,19 @@
 import { ADD_APD_YEAR, EDIT_APD, REMOVE_APD_YEAR } from './symbols';
 import { updateBudget } from '../budget';
 
+/**
+ * Rename an activity
+ * @param {Number} index The index of the activity to rename
+ * @param {String} name The new activity name
+ */
+ export const setActivityName = (index, name) => (dispatch, getState) => {
+  dispatch({
+    type: EDIT_APD,
+    path: `/apd/${index}/name`,
```

Comment:
I see what's happening. In this case, path is a json path. We are using jsonpatch http://jsonpatch.com/. This dispatch is setting out an EDIT_APD message which is picked up by the apd reducer, which updates that path in the json with the value you sent in.